src/Core/Services/TokenService.cs

```diff
@@ -64,6 +78,24 @@ public async Task<string> GetRefreshTokenAsync()
             return _refreshToken;
         }
 
+        public async Task ToggleTokensAsync()
+        {
+            var token = await GetTokenAsync();
+            var refreshToken = await GetRefreshTokenAsync();
+            var timeout = await _storageService.GetAsync<int?>(Constants.VaultTimeoutKey);
+            var action = await _storageService.GetAsync<string>(Constants.VaultTimeoutActionKey);
+            if ((timeout != null || timeout == 0) && action == "logOut")
```

Comment:
Is this logic intended? if `timeout != null` wouldn't zero (`0`) be included in that test? Also, should use `timeout.HasValue` instead of `!= null` (IMO)

Reply:
Great point - thanks for brining it up! I mentioned on my jslib PR regarding this exact check. I do agree with your suggestions. If we do make the change, I can use the private method `SkipTokenStorage` instead of the explicit checks here.

Reply:
Updated PR.